src/Pos/ForwardClient/Client.hs

```diff
@@ -42,10 +42,10 @@ getAgentID agent = do
 -- The name of the upload is defaulted to logs.log
 uploadLogs :: Agent -> [(FilePath, LByteString)] -> IO Token
 uploadLogs agent logs = do
-    BSL.writeFile "logs.log" $ joinLogs logs
+    BSL.writeFile "logs.zip" $ joinLogs logs
```

Comment:
Wait for it, it doesn't make sense.
1. Use `withSystemTempFile` or something. See how `withTempLogFile` is implemented in CSL: https://github.com/input-output-hk/cardano-sl/blob/develop/infra/Pos/Reporting/Methods.hs#L179
2. This is not a `zip` file, so it shouldn't have `.zip` extension. It's not hard to actually zip something on the report server, but why do it on the server in the first place? CSL send tar.lzma archive, for example. See `compressLogs` in the link above. I recommend using tar on the report server and compress data on the client's machine to avoid server load.

Reply:
I've agreed on (2) (though it's anyway a super dirty hack), but (1) is still relevant.

Reply:
Daedalus compresses and after that splits the logs into chunks (it's not a single file, but a directory) 
I'm not sure I understand how `withTempLogFile` can replace this.

Reply:
Well, instead of writing to a real file every time, write to a temporary one.

Reply:
Any progress?

Reply:
I'm on it